src/App.jsx

```diff
@@ -1,5 +1,46 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import RegionsContainer from './RegionsContainer';
+import CategoriesContainer from './CategoriesContainer';
+import RestaurantsContainer from './RestaurantsContainer';
+
+import {
+  loadButtonData,
+  loadFilteredRestaurants,
+  applyFilter,
+} from './actions';
+
 export default function App() {
+  const { filter } = useSelector((state) => ({
+    filter: state.filter,
+  }));
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(loadButtonData('regions'));
+    dispatch(loadButtonData('categories'));
+  }, []);
+
+  useEffect(() => {
+    if (!(filter.regionName && filter.categoryId)) return;
+
+    dispatch(loadFilteredRestaurants({
+      regionName: filter.regionName,
+      categoryId: filter.categoryId,
+    }));
+  }, [filter]);
+
+  function handleClick({ field, content }) {
+    dispatch(applyFilter({ field, content }));
+  }
```

Comment:
각각 Container에서 관리하지 않고 App에서 관리해주시는 특별한 이유가 있으신가용?ㅎㅎ

Reply:
```suggestion
  useEffect(() => {
    if (!(filter.regionName && filter.categoryId)) return;

    dispatch(loadFilteredRestaurants({
      regionName: filter.regionName,
      categoryId: filter.categoryId,
    }));
  }, [filter]);
```

이 부분의 경우에는 고민이 좀 있었는데요! restaurants이라는 정보를 받아오긴 하지만 filter라고 정해놓은 state를 관리하는 건 app에서 하는게 맞지 않나라고 생각했습니다!

Reply:
```
  function handleClick({ field, content }) {
    dispatch(applyFilter({ field, content }));
  }
```

handleClick 함수는 CategoriesContainer와 RegionsContainer 모두에 같이 사용(fetchButtonData로 fetch 같은 함수 사용)해서 App에서 관리헤줬습니다!